File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Create an array...
        arr = []
        arr.append(nums[0])
        # Initialize the max sum...
        maxSum = arr[0]
        # Traverse all the element through the loop...
        for i in range(1, len(nums)):
            # arr[i - 1] + nums[i](largest sum plus current number with suing prefix)
            # calculare arr[0], arr[1]..., arr[n] while comparing each one with current largest sum...
            arr.append(max(arr[i - 1] + nums[i], nums[i]))
            if arr[i] > maxSum:
                maxSum = arr[i]
        return maxSum
        

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Initialize currMaxSum & take first element of array from which we start tp do sum...
        int maxSum = nums[0];
        // Initialize the current sum of our subarraay as nums[0]...
        int currSum = nums[0];
        // Traverse all the element through the loop...
        for(int i = 1; i < nums.length; i++) {
            // Do sum of elements contigous with curr sum...
            // Compare it with array element to get maximum result...
            currSum = Math.max(currSum + nums[i], nums[i]);
            // Compare current sum add max sum.
            maxSum = Math.max(maxSum, currSum);
        }
        return maxSum;
    }
}